import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { catchError, Observable, retry, throwError } from 'rxjs';
import { Cat } from '../model/Cat';

@Injectable({
  providedIn: 'root'
})
export class CatsService {

  constructor(private http: HttpClient) { }

  baseurl = 'https://freetestapi.com/api';

  // GET
  getCats(): Observable<Cat>{
    return this.http.get<Cat>(this.baseurl + '/v1/cats').pipe(retry(1),catchError(this.errorHandl))
  }

  errorHandl(error:any){
    let errorMessage = ''
    // o erro estÃ¡ vindo do front
    if (error.error instanceof ErrorEvent){
      errorMessage = error.error.message;
    }else{
      // erro que veio da API
      errorMessage = `Error code: ${error.status} \n Message: ${error.message}`
    }
    console.log(errorMessage);
    return throwError(() => {
      return errorMessage
    })
  }



}
